lib/membrane_h264_plugin/parser.ex

```diff
@@ -230,20 +228,11 @@ defmodule Membrane.H264.Parser do
 
   @impl true
   def handle_end_of_stream(:input, ctx, state) when state.mode != :au_aligned do
-    {last_nalu_payload, nalu_splitter} = NALuSplitter.flush(state.nalu_splitter)
-
-    {{access_units, au_splitter}, nalu_parser} =
-      if last_nalu_payload != <<>> do
-        {last_nalu, nalu_parser} = NALuParser.parse(last_nalu_payload, state.nalu_parser)
-        {AUSplitter.split([last_nalu], state.au_splitter), nalu_parser}
-      else
-        {{[], state.au_splitter}, state.nalu_parser}
-      end
-
-    {last_au, au_splitter} = AUSplitter.flush(au_splitter)
-    maybe_improper_aus = access_units ++ [last_au]
-
-    {actions, state} = prepare_actions_for_aus(maybe_improper_aus, state)
+    {last_nalu_payload, nalu_splitter} = NALuSplitter.split(<<>>, true, state.nalu_splitter)
+    {last_nalu, nalu_parser} = NALuParser.parse_nalus(last_nalu_payload, state.nalu_parser)
+    {maybe_improper_aus, au_splitter} = AUSplitter.split(last_nalu, true, state.au_splitter)
```

Comment:
I don't like the fact that we are invoking `AUSplitter.split` with `is_au_aligned=true`  in case, where the `state.mode != :au_aligned`.  Perhaps we could call the argument differently, e.g. `should_flush`?

Reply:
done